homeassistant/components/sensorpush/data.py

```diff
@@ -0,0 +1,87 @@
+"""The SensorPush Bluetooth integration."""
+from __future__ import annotations
+
+from sensorpush_ble import (
+    ATTR_MANUFACTURER as SENSOR_MANUFACTURER,
+    ATTR_MODEL as SENSOR_MODEL,
+    ATTR_NAME as SENSOR_NAME,
+    SIGNAL_STRENGTH_KEY,
+    DeviceClass,
+    DeviceKey,
+    SensorDeviceInfo,
+    SensorUpdate,
+)
+
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+)
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntityDescription
+from homeassistant.const import ATTR_MANUFACTURER, ATTR_MODEL, ATTR_NAME
+from homeassistant.helpers.entity import DeviceInfo
+
+SENSOR_DEVICE_CLASS_TO_HASS = {
+    DeviceClass.BATTERY: SensorDeviceClass.BATTERY,
+    DeviceClass.HUMIDITY: SensorDeviceClass.HUMIDITY,
+    DeviceClass.TEMPERATURE: SensorDeviceClass.TEMPERATURE,
+    DeviceClass.PRESSURE: SensorDeviceClass.PRESSURE,
+    DeviceClass.SIGNAL_STRENGTH: SensorDeviceClass.SIGNAL_STRENGTH,
+}
+
+
+def _device_key_to_bluetooth_entity_key(
+    device_key: DeviceKey,
+) -> PassiveBluetoothEntityKey:
+    """Convert a device key to an entity key."""
+    return PassiveBluetoothEntityKey(device_key.key, device_key.device_id)
+
+
+def _sensor_device_class_to_hass(
+    sensor_device_class: DeviceClass | None,
+) -> SensorDeviceClass | None:
+    """Convert a sensor device class to a sensor device class."""
+    if sensor_device_class is None:
+        return None
+    return SENSOR_DEVICE_CLASS_TO_HASS.get(sensor_device_class)
+
+
+def _sensor_device_info_to_hass(
+    device_info: SensorDeviceInfo,
+) -> DeviceInfo:
+    """Convert a sensor device info to a sensor device info."""
+    base_device_info = DeviceInfo({})
```

Comment:
Both the input and output are called `device_info`. Can we name it sensor_info -> hass_device_info maybe?

Reply:
d1cca09dbe26b16b5d376397775191deb67b0110